File: samples/AdventureWorksLT/UdtNameStyleAttribute.cs
using DevZest.Data;
using DevZest.Data.Addons;
using DevZest.Data.Annotations.Primitives;
using System;

namespace DevZest.Samples.AdventureWorksLT
{
    [ModelDesignerSpec(addonTypes: new Type[] { typeof(ColumnNotNull) }, validOnTypes: new Type[] { typeof(_Boolean) })]
    public sealed class UdtNameStyleAttribute : ColumnAttribute
    {
        protected override void Wireup(Column column)
        {
            if (column is _Boolean boolean)
                boolean.Nullable(false);
        }
    }
}

File: samples/AdventureWorksLT.VB/UdtNameStyleAttribute.vb
Imports DevZest.Data
Imports DevZest.Data.Annotations.Primitives
Imports DevZest.Data.Addons

Namespace DevZest.Samples.AdventureWorksLT
    <ModelMemberAttributeSpec(GetType(ColumnNotNull), True, GetType(_Boolean))>
    Public NotInheritable Class UdtNameStyleAttribute
        Inherits ColumnAttribute

        Protected Overrides Sub Wireup(column As Column)
            If TypeOf column Is _Boolean Then
                column.Nullable(False)
            End If
        End Sub
    End Class
End Namespace
